smart-contract-verifier/smart-contract-verifier/src/lookup_methods/find_methods.rs

```diff
@@ -0,0 +1,139 @@
+use super::{
+    disassemble::{disassemble_bytecode, DisassembledOpcode},
+    method::Method,
+};
+use bytes::Bytes;
+use ethers_core::abi::Abi;
+use ethers_solc::sourcemap::SourceMap;
+use std::{collections::BTreeMap, iter::repeat};
+
+pub struct LookupMethodsRequest {
+    pub bytecode: Bytes,
+    pub abi: Abi,
+    pub source_map: SourceMap,
+    pub file_ids: BTreeMap<u32, String>,
+}
+
+pub struct LookupMethodsResponse {
+    pub methods: BTreeMap<String, Method>,
+}
+
+pub fn find_methods(request: LookupMethodsRequest) -> LookupMethodsResponse {
+    let methods = parse_selectors(request.abi);
+    let opcodes = disassemble_bytecode(&request.bytecode);
+    let opcodes = opcodes.as_slice();
+
+    let methods = methods
+        .into_iter()
+        .filter_map(|(func_sig, selector)| {
+            let func_index = match find_src_map_index(&selector, opcodes) {
```

Comment:
Nit: then the `opcodes` could be a vector (and there would be no need in `opcodes.as_slice()` line)
```suggestion
            let func_index = match find_src_map_index(&selector, &opcodes) {
```